src/renderer/VectorSkin.js

```diff
@@ -37,22 +57,35 @@ export default class VectorSkin extends Skin {
 
     // Prevent IndexSizeErrors if the image is too small to render
     if (width === 0 || height === 0) {
-      this._mipmaps.set(mipLevel, null);
-      return;
+      return null;
     }
 
+    // Instead of uploading the image to WebGL as a texture, render the image to a canvas and upload the canvas.
+    const canvas = this._canvas;
+    const ctx = canvas.getContext("2d");
+
     canvas.width = width;
     canvas.height = height;
 
     ctx.drawImage(image, 0, 0, width, height);
+    return this._canvas;
+  }
 
-    // Use linear (i.e. smooth) texture filtering for vectors
-    this._mipmaps.set(mipLevel, this._makeTexture(canvas, this.gl.LINEAR));
+  // TODO: handle proper subpixel positioning when SVG viewbox has non-integer coordinates
+  // This will require rethinking costume + project loading probably
```

Comment:
I'd love to hear details here, probably put into an issue if this'll be something to address later!

Reply:
I wrote up a summary of this a while back, over in https://github.com/LLK/scratch-render/issues/550. There's an associated PR that fixes it in https://github.com/LLK/scratch-render/pull/594.

I can't remember what it has to do with costume loading--in general, I'm not really happy with the whole "costumes pop into existence when they're loaded" way things currently work in Leopard. I'd prefer if there was some sort of "preload" function that you called before running a project that loaded all the costumes and sounds.

Reply:
I'm not as experienced with Leopard or Scratch or the differences between them, but I definitely got the same feeling while reviewing your code. Maybe you could open an issue suggesting a "preload" mechanic? I feel like proper support for that (alongside a "progress bar" for asset loading etc) would be pretty reasonable/good.

Reply:
I've opened two issues branching off this:

* #132 
* #133